### PR TITLE
Optimizer pass options

### LINKED ISSUE #561: Allow for individual enabeling / disabling of passes
Allow the optimizer to enable / disable independent groups of passes one by one

### PR DESCRIPTION
**Pull request by jdahm**
## Technical Description

Add options to turn on individual passes, which can be enabled after passing `-default-none`.

### Resolves / Enhances

Resolves #561.

### CODE REVIEW

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
@twicki @havogt this is failing because it's building against the wrong clang-gridtools branch. As far as I can tell, "build with parameters" only let's me build branches (or sha hashes) that exist in the MeteoSwiss-APN/dawn repo. The problem is that the branch for the PR is in my fork.

**Comment by jdahm** [comment]
@havogt told me that hashes from PRs work, even if the branch isn't in the main dawn repo.

**Comment by jdahm** [comment]
This passes on Jenkins with a manual launch at `job/dawn_gtclang_clang-gridtools_PR/118/`

**Comment by jdahm** [comment]
@twicki Could you take a look at this PR before I squash and merge to `optimizer_refactor`?

File: dawn/src/dawn/Compiler/DawnCompiler.cpp [hunk 1]
```diff
@@ -91,6 +91,10 @@ struct ComputeEditDistance<std::string> {
   }
 };
 
+bool shouldRunPass(const Options& options, bool runSpecificPass) {
```
Comment (twicki) [inline dawn/src/dawn/Compiler/DawnCompiler.cpp]: if we use it like this there is no real reason to do a `checkAndPushBack` right? Would we not want to push back directly then?

Comment (jdahm) [reply dawn/src/dawn/Compiler/DawnCompiler.cpp]: True, and checkAndPushBack and the whole PassManager is being removed in another PR :)

**Pull request merged**